### PR TITLE
update method for reading templates

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek <rudrakshpareek3601@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**
Updates `template.String()` to use `ReadFileSource()` for reading templates, which supports multiple URI schemes.

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — APPROVED** [review]
LGTM

**Comment by leecalcote** [comment]
Are there any considerations necessary to be made before merging and releasing this, @layer5io/adapter-maintainers?

**Comment by kumarabd** [comment]
> Are there any considerations necessary to be made before merging and releasing this, @layer5io/adapter-maintainers?

Looks good.

**Pull request merged**
